server/xcode/xcode_darwin.go

```diff
@@ -78,11 +79,15 @@ func (x *xcodeLocator) xcodeVersionForVersionString(version string) *xcodeVersio
 
 // Locates all all XCode versions installed on the host machine.
 // Very losely based on https://github.com/bazelbuild/bazel/blob/master/tools/osx/xcode_locator.m
-func (x *xcodeLocator) locate() {
+func (x *xcodeLocator) locate() error {
 	bundleID := stringToCFString(xcodeBundleID)
 	defer C.CFRelease(C.CFTypeRef(bundleID))
 
 	urlsPointer := C.LSCopyApplicationURLsForBundleIdentifier(bundleID, nil)
+
+	if urlsPointer == C.CFArrayRef(unsafe.Pointer(nil)) {
+		return fmt.Errorf("cannot find XCode bundle: %q", xcodeBundleID)
```

Comment:
Could we use something from status instead of minting a new fmt.Errorf? FailedPreconditionError seems like it would fit

Reply:
could also give the user a hint like - Make sure Xcode is installed

Reply:
nit: It's Xcode not XCode

Reply:
All Done.